### PR TITLE
Replace button class with optimized template (flash savings)

### PR DESCRIPTION
**Pull request by CapnBry**
This is a refactor of the lib/BUTTON class with an optimized version that has roughly the same functionality but uses about 528 bytes less code and 4 bytes less RAM. The new code also has slightly less MCU cycle overhead because it only samples the button every 25ms instead of as fast as possible with the old code.

### Example Code
```
Button <PB2, false> button;
...
// in setup()
button.OnShortPress = []() { if (button.getCount() == 3) Serial.println("TRIPPLEEEE"); };
button.OnLongPress = []() { Serial.print("LONG"); };
...
// in loop()
button.update();
```

### Similarities
There's still a short press and long press event that function identically to the old code, as far as my testing and pressing a button about 100,000 times this afternoon can tell. There's still a debounce. Long still has a repeat so there's a long event every 500ms, starting 500ms after the button goes down. And Long still blocks the short event.

### Regressions
I've changed the API to it so there's no longer a specialized TriplePress handler. Instead, there's a `getCount()` method that the receiver can use to count any number of presses. The old code also had variables for all the timing, but there was no way to change them from the defaults, so I did not make them changeable either.

### CODE REVIEW

**Comment by JyeSmith** [comment]
@LinKR00001 can you please test for BETAFPV and provide feedback.

File: src/lib/BUTTON/button.h [hunk 1]
```diff
@@ -1,37 +1,89 @@
...
-    static bool buttonIsDownLong; //is the button currently being held down?
+    // State
+    uint32_t _lastCheck;  // millis of last pin read
+    uint32_t _lastFallingEdge; // millis of last debounced falling edge
+    uint8_t _state; // pin history
+    bool _isLongPress; // true if last press was a long
+    uint8_t _pressCount; // number of short presses before timeout
+public:
+    // Callbacks
+    std::function<void ()>OnShortPress;
```
Comment (cruwaller) [inline src/lib/BUTTON/button.h:27]: How about if `_pressCount` is argument of OnShortPress callback?

Comment (CapnBry) [reply src/lib/BUTTON/button.h:27]: I had it this way originally, but passing the count to the handler adds 20 bytes of code to all platforms whether they use the count or not. The BetaFPV 2400 TX target uses the same code space if the value is passed or not so it made sense to not pass it despite being a less desirable pattern.

**Comment by LinKR00001** [comment]
We have already tested it,It work fine.


> @LinKR00001 can you please test for BETAFPV and provide feedback.

**Pull request merged**